### PR TITLE
Golangci

### PR DESCRIPTION
**Pull request by whites11**
towards: https://github.com/ClickHouse/terraform-provider-clickhouse/issues/95

address golangci-lint warnings and run it automatically during PRs to avoid unformatted code from being merged

### CODE REVIEW

File: clickhouse/service.go [hunk 1]
```diff
@@ -764,19 +765,14 @@ func (r *ServiceResource) Update(ctx context.Context, req resource.UpdateRequest
...
-		res, err := r.client.UpdateServicePassword(serviceId, ServicePasswordUpdateFromPlainPassword(password))
+		_, err := r.client.UpdateServicePassword(serviceId, ServicePasswordUpdateFromPlainPassword(password))
 		if err != nil {
 			resp.Diagnostics.AddError(
 				"Error Updating ClickHouse Service Password",
 				"Could not update service password, unexpected error: "+err.Error(),
 			)
 			return
 		}
-
```
Comment (whites11) [inline clickhouse/service.go:775]: assigned but never read afterwards

File: clickhouse/service.go [hunk 2]
```diff
@@ -790,22 +786,24 @@ func (r *ServiceResource) Update(ctx context.Context, req resource.UpdateRequest
...
-		// empty password provided, ClickHouse Cloud return a new generated password
-		if len(res.Password) > 0 {
```
Comment (whites11) [inline clickhouse/service.go:803]: never read afterwards

File: clickhouse/utils.go [hunk 3]
```diff
@@ -3,46 +3,13 @@ package clickhouse
...
 	"github.com/hashicorp/terraform-plugin-framework/types"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
 func createEmptyList(listType attr.Type) types.List {
 	emptyList, _ := types.ListValue(listType, []attr.Value{})
 	return emptyList
 }
 
-func diffArrays[T any, V comparable](a []T, b []T, hash func(T) V) ([]T, []T) {
```
Comment (whites11) [inline clickhouse/utils.go:14]: unused

File: clickhouse/utils.go [hunk 4]
```diff
@@ -54,8 +21,3 @@ func equal[T comparable](a []T, b []T) bool {
 	}
 	return true
 }
-
-func resetValue(d *schema.ResourceData, key string) {
```
Comment (whites11) [inline clickhouse/utils.go:58]: unused

File: main.go [hunk 5]
```diff
@@ -11,7 +11,10 @@ import (
 //go:generate go run github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs generate --provider-name clickhouse
 
 func main() {
-	providerserver.Serve(context.Background(), clickhouse.New, providerserver.ServeOpts{
+	err := providerserver.Serve(context.Background(), clickhouse.New, providerserver.ServeOpts{
 		Address: "clickhouse.cloud/terraform/clickhouse",
 	})
+	if err != nil {
+		panic(err)
```
Comment (nikita-vanyasin) [inline main.go:18]: Maybe just print error and exit(1) ?
Not sure if panic is handled properly by code which executes the provider

Comment (whites11) [reply main.go]: let me just ignore the error check. I checked some official provider code and people doesn't check that err

**Pull request merged**